src/machine/asm/execute.S

```diff
@@ -2191,6 +2197,37 @@ ckb_vm_x64_execute:
   WRITE_RS3(RS2r)
   NEXT_INST
 .p2align 3
+.CKB_VM_ASM_LABEL_OP_ADC:
```

Comment:
This is a direct translation of the 5 instructions from RISC-V, but from a semantic point of view, what we need to do here, is simply:

* Set or clear carry flag based on passed value from RISC-V register(there are many [ways](https://stackoverflow.com/questions/36798572/how-can-i-set-or-clear-overflow-flag-in-x86-assembly) to do it)
* Use adc on the 2 source RISC-V registers
* Move carry flag to the output register

This flow, IMO, really matches what ADC is really doing.

Reply:
I can't get the correct a2 value under the steps you recommend. Consider the result of a2, I think direct translation is the most convenient. 

```
a2 = (tmp + a2) 's carry flag
where tmp = a0 + a1
```